pkg/slareport/s3.go

```diff
@@ -0,0 +1,70 @@
+package slareport
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// PDFUploader uploads pdfs to the configured S3 endpoint.
+type PDFUploader struct {
+	client *minio.Client
+	bucket string
+	ctx    context.Context
```

Comment:
Please don't save the context in the struct. You should pass it when calling `Upload`.